src/02-room-details.js

```diff
@@ -25,7 +25,31 @@ const exampleRoomData = require("../data/rooms");
  *  getRoomByDinosaurName(dinosaurs, rooms, "Pterodactyl");
  *  //> "Dinosaur with name 'Pterodactyl' cannot be found."
  */
-function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
+
+function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {
+  let dinoId = "";
+  let dinoRoom = "";
+
+  for (let i = 0; i < dinosaurs.length; i++) {
+    if (dinosaurs[i].name === dinosaurName) {
+      dinoId = dinosaurs[i].dinosaurId;
+    }
+  }
+
+  for (let j = 0; j < rooms.length; j++) {
```

Comment:
techincally we don't need our index to be `j` we could initialize our for loop with `let i =0` just like above since we are not in the scope of the prior for loop.